src/controllers/controller.scatter.js

```diff
@@ -1,7 +1,126 @@
-import LineController from './controller.line';
+import DatasetController from '../core/core.datasetController';
+import {isNullOrUndef} from '../helpers';
+import {isNumber} from '../helpers/helpers.math';
+import {_getStartAndCountOfVisiblePoints, _scaleRangesChanged} from '../helpers/helpers.extras';
+import registry from '../core/core.registry';
 
-export default class ScatterController extends LineController {
+export default class ScatterController extends DatasetController {
+  initialize() {
+    const {showLine} = this.getDataset();
```

Comment:
Not sure if this also needs to get checked from the options object itself here

Reply:
We can't check the showLine parameter from the Options object because the Options object is created after the controller initialize. 
In this case, we create the wrong controller without a datasetElementType and then initialize the correct one. This is the wrong solution if we can immediately initialize the suitable controller.

Reply:
fixed